pkg/platform/kube/logProxy/elastic/opensearch.go

```diff
@@ -48,23 +53,249 @@ func NewOpenSearchLogProxy(config *platformconfig.ElasticSearchConfig) (*OpenSea
 		// Set to true to skip TLS verification if SSLVerificationMode is "none"
 		InsecureSkipVerify: config.SSLVerificationMode == "none",
 	}
-	if openSearchClient.client, err = opensearch.NewClient(opensearch.Config{
+	if openSearchClient.client, err = opensearchapi.NewClient(opensearchapi.Config{Client: opensearch.Config{
 		Addresses: []string{config.URL},
 		Password:  config.Password,
 		Username:  config.Username,
 		Transport: &http.Transport{
 			TLSClientConfig: tlsConfig,
 		},
-	}); err != nil {
+	}}); err != nil {
 		return nil, errors.Wrap(err, "Failed to create opensearch client")
 	}
+
 	return openSearchClient, err
 }
 
+func (o *OpenSearchLogProxy) GetFunctionReplicas(ctx context.Context, options *logProxy.GetFunctionReplicaOptions) ([]string, error) {
+	if o.client == nil {
+		return nil, errors.New("OpenSearch client is not configured")
+	}
+	query := o.getFunctionBaseSearchRequest(options.FunctionName)
+	query["aggs"] = map[string]interface{}{
+		"distinct_pod_names": map[string]interface{}{
+			"terms": map[string]interface{}{
+				"field": "kubernetes.pod.name",
```

Comment:
"kubernetes.pod.name" will now repeat in 6 places (elastic + opensearch), so we can create a const for it